### DISCUSSION TITLE
Select() equivalent in MODULE.bazel

### DISCUSSION THREAD

**Discussion by ScottHo** [bazelbuild/bazel#24121] [Q&A]
Is it possible to have some sort of switch in MODULE.bazel files?

I'm using conan, and have different use_repo() calls for linux and windows

Is there a way to toggle between these two calls?
`use_repo(load_conan_dependencies, "linux_lib", "common_lib", ...)`
`use_repo(load_conan_dependencies, "windows_lib", "common_lib", ...)`

**Comment by meteorcloudy** [comment]
The current answer is no. There is no conditional dependency support in MODULE.bazel. You could have the `load_conan_dependencies` module extension generate different repo for the same repo name and mark it as [`os_dependent`](https://bazel.build/rules/lib/globals/bzl#parameters_6) and use 
```
use_repo(load_conan_dependencies, "platform_specific_lib", "common_lib", ...)
```
But then your lockfile has to be updated separately on each platfrom.

**Reply by fmeum** [reply]
If possible, try to generate repos for each platform regardless of what the host machine is and then use `select` in BUILD files. This is much more flexible and also works better with multi-platform builds.

**Reply by ScottHo** [reply]
Thanks for the replies. What I ended up doing is generating a MODULE.bazel when I generate the dependencies form conan, and using the import() statement
